src/domain/chains/entities/schemas/chain.schema.ts

```diff
@@ -1,28 +1,118 @@
   },
   required: ['name', 'symbol', 'decimals', 'logoUri'],
 };
 
-const chainSchema: JSONSchemaType<Chain> = {
+export const rpcUriSchema: JSONSchemaType<RpcUri> = {
+  type: 'object',
+  properties: {
+    authentication: {
+      type: 'string',
+      enum: Object.values(RpcUriAuthentication),
+    },
+    value: { type: 'string' },
+  },
+  required: ['authentication', 'value'],
+};
+
+export const blockExplorerUriTemplateSchema: JSONSchemaType<BlockExplorerUriTemplate> =
+  {
+    type: 'object',
+    properties: {
+      address: { type: 'string' },
+      txHash: { type: 'string' },
+      api: { type: 'string' },
+    },
+    required: ['address', 'txHash', 'api'],
+  };
+
+export const themeSchema: JSONSchemaType<Theme> = {
+  type: 'object',
+  properties: {
+    textColor: { type: 'string' },
+    backgroundColor: { type: 'string' },
+  },
+  required: ['textColor', 'backgroundColor'],
+};
+
+export const gasPriceSchema: JSONSchemaType<
+  Array<GasPriceOracle | GasPriceFixed>
+> = {
+  type: 'array',
+  items: {
+    anyOf: [
+      {
+        type: 'object',
+        properties: {
+          type: { const: 'fixed', type: 'string' },
+          weiValue: { type: 'string' },
+        },
+        required: ['type', 'weiValue'],
+      },
+      {
+        type: 'object',
+        properties: {
+          type: { const: 'oracle', type: 'string' },
+          uri: { type: 'string', format: 'uri' },
+          gasParameter: { type: 'string' },
+          gweiFactor: { type: 'number' },
+        },
+        required: ['type', 'uri', 'gasParameter', 'gweiFactor'],
+      },
+    ],
+  },
+};
+
+export const chainSchema: JSONSchemaType<Chain> = {
   type: 'object',
   properties: {
     chainId: { type: 'string' },
     chainName: { type: 'string' },
-    transactionService: { type: 'string' },
-    vpcTransactionService: { type: 'string' },
-    nativeCurrency: { $ref: 'nativeCurrency' },
+    shortName: { type: 'string' },
+    rpcUri: { $ref: 'rpcUriSchema' },
+    safeAppsRpcUri: { $ref: 'rpcUriSchema' },
+    publicRpcUri: { $ref: 'rpcUriSchema' },
+    blockExplorerUriTemplate: { $ref: 'blockExplorerUriTemplateSchema' },
+    nativeCurrency: { $ref: 'nativeCurrencySchema' },
+    transactionService: { type: 'string', format: 'uri' },
+    vpcTransactionService: { type: 'string', format: 'uri' },
+    theme: { $ref: 'themeSchema' },
+    gasPrice: { $ref: 'gasPriceSchema' },
+    ensRegistryAddress: { type: 'string' },
+    disabledWallets: { type: 'array', items: { type: 'string' } },
+    features: { type: 'array', items: { type: 'string' } },
   },
-  required: ['chainId', 'chainName', 'transactionService'],
+  required: [
+    'chainId',
+    'chainName',
+    'shortName',
+    'rpcUri',
+    'safeAppsRpcUri',
+    'publicRpcUri',
+    'blockExplorerUriTemplate',
+    'nativeCurrency',
+    'transactionService',
+    'vpcTransactionService',
+    'theme',
+    'gasPrice',
+    'ensRegistryAddress',
```

Comment:
Maybe shouldn't be required

Reply:
Done in [34d40f2](https://github.com/5afe/safe-client-gateway-nest/pull/93/commits/34d40f226f67c3b006a3f1eccccfe9dc022ad746)